### DISCUSSION TITLE
Cannot open include file fmt/format.h

### DISCUSSION THREAD

**Discussion by Kaaserne** [xmake-io/xmake#3336] [Q&A] [answered]
```xmake
add_requires("catch2 v2.13.10", "fmt >=7")

target("a")
    set_kind("headeronly")
    add_includedirs("include", { interface = true })
    add_packages("fmt")

target("a tests")
    set_kind("binary")
    add_files("tests/*.cpp")
    add_deps("a")
    -- add_packages("fmt") fixes the problem, but shouldn't, since target a already contains this
    add_packages("catch2")

package("cpp-lazy")
    set_description("A fast C++11/14/17/20 header only library for lazy evaluation and function tools")
    add_deps("cpp-lazy")
```
When building target `a tests`, a header file in target `a` generates the following error: `fatal error C1083: Cannot open include file: 'fmt/format.h': No such file or directory`. Adding `add_packages("fmt")` to target `a tests` solves the problem, but seems wrong.

**Comment by waruqi** [comment, marked as answer]
set `{public = true}` in target a

`add_packages("fmt", {public = true})`

**Reply by Kaaserne** [reply]
Is this something that was in the docs? Could not find it [here](https://xmake.io/#/manual/project_target?id=targetadd_packages)

**Reply by waruqi** [reply]
https://xmake.io/#/manual/project_target?id=add-target-dependencies
